Bugzilla/App.pm

```diff
@@ -122,6 +122,16 @@ sub startup {
         else {
           $c->res->headers->cache_control('public, max-age=31536000, immutable');
         }
+        if ($self->https
+          && not $c->res->headers->strict_transport_security
```

Comment:
Here either you should use only the C-style logical operations (`&& || !`) or only the python ones (`and or not`) otherwise the results can be surprising.

Reply:
Also this entire `if` should be added to a new `after_dispatch` hook, possible outside the `if ($self->mode ne 'development') {` conditional/

```perl
$self->hook(after_dispatch => sub {
my ($c) = @_;
# <insert your code>
});
```